jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/LambdaBuilderUtils.kt

```diff
@@ -0,0 +1,134 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda
+
+import com.intellij.build.BuildProgressListener
+import com.intellij.build.BuildViewManager
+import com.intellij.build.DefaultBuildDescriptor
+import com.intellij.build.events.impl.FailureResultImpl
+import com.intellij.build.events.impl.FinishBuildEventImpl
+import com.intellij.build.events.impl.OutputBuildEventImpl
+import com.intellij.build.events.impl.StartBuildEventImpl
+import com.intellij.build.events.impl.SuccessResultImpl
+import com.intellij.execution.process.ProcessAdapter
+import com.intellij.execution.process.ProcessEvent
+import com.intellij.execution.process.ProcessListener
+import com.intellij.execution.process.ProcessOutputTypes
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.application.runInEdt
+import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.progress.Task
+import com.intellij.openapi.util.Key
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.CompletionStage
+
+object LambdaBuilderUtils {
+    fun buildAndReport(
+        module: Module,
+        runtimeGroup: RuntimeGroup,
+        request: BuildLambdaRequest
+    ): CompletionStage<BuiltLambda> {
+        val future = CompletableFuture<BuiltLambda>()
+        val project = module.project
+
+        val buildViewManager = ServiceManager.getService(project, BuildViewManager::class.java)
+
+        runInEdt(ModalityState.NON_MODAL) {
+            val descriptor = DefaultBuildDescriptor(request, "Run SAM build", "", System.currentTimeMillis())
```

Comment:
workingDir should probably be set correctly here

Reply:
Yep, will use content root of module